src/common.cpp

```diff
@@ -232,20 +231,20 @@ namespace irods::http
         log::debug("{}: Bearer token: [{}]", __func__, bearer_token);
 
         // Verify the bearer token is known to the server. If not, return an error.
-        const auto* object = irods::process_stash::find(bearer_token);
-        if (!object) {
-            log::error("{}: Could not find bearer token matching [{}].", __func__, bearer_token);
+        const auto object{irods::http::process_stash::find(bearer_token)};
+		try {
+		  const auto p{boost::any_cast<const authenticated_client_info&>(object)};
+		  if (std::chrono::steady_clock::now() >= p.expires_at) {
+            log::error("{}: Session for bearer token [{}] has expired.", __func__, bearer_token);
             return {.response = fail(status_type::unauthorized)};
-        }
+		  }
+          log::trace("{}: Client is authenticated.", __func__);
+		  return {.client_info = p};
```

Comment:
I think if we were to not `const` the variable `p`, we could do a move on the final line?

Reply:
Go for it.